SpringBoot 3.0: Create FATs

tbitonti opened the pull request:
Creation of 30 FAT as copies of 20 FATs.
Several test driven fixes;
FeatureAuditor updates for SB3 cases (still in progress). Partially unified spring version dependencies;
.classpath and .gitignore cleanup.
Added SB3 server config template.

---

dev/cnf/oss_dependencies.maven
-- Updates for modified Spring versions
dev/cnf/resources/bnd/bundle.spring.props
-- Table of Spring versions
dev/wlp-gradle/bndSettings.gradle
-- Updates for modified Spring versions
dev/wlp-gradle/subprojects/spring.gradle
-- Table of Spring versions

dev/com.ibm.ws.app.manager.springboot/bnd.bnd
dev/com.ibm.ws.springboot.support.shutdown/bnd.bnd
dev/com.ibm.ws.springboot.support.web.server/bnd.bnd
dev/com.ibm.ws.springboot.support.web.server.version15/bnd.bnd
dev/com.ibm.ws.springboot.support.web.server.version20/bnd.bnd
dev/io.openliberty.springboot.support.web.server.version30/bnd.bnd
-- Update to use spring version imports.

dev/com.ibm.ws.springboot.support.shutdown/src/com/ibm/ws/springboot/support/shutdown/FeatureAuditor.java
-- Update to for new SB 3.0 cases.

dev/io.openliberty.springboot.support.web.server.version30/src/io/openliberty/springboot/support/web/server/version30/container/LibertyReactiveWebServerFactory.java
-- Update version ranges from "[3.0.0, 3.0.4[" to "[3.0.0, 4.0.0[".

dev/build.image/publish/wlp/templates/servers/springBoot3/server.xml
-- Added template for SpringBoot3

com.ibm.ws.springboot.fat20.actuator.app
-- rename of com.ibm.ws.springboot.support.version20.test.actuator.app
com.ibm.ws.springboot.fat20.app
-- rename of com.ibm.ws.springboot.support.version20.test.app
com.ibm.ws.springboot.fat20.java.app
-- rename of com.ibm.ws.springboot.support.version20.test.java.app
com.ibm.ws.springboot.fat20.multicontext.app
-- rename of com.ibm.ws.springboot.support.version20.test.multi.context.app
com.ibm.ws.springboot.fat20.war.app
-- rename of com.ibm.ws.springboot.support.version20.test.war.app
com.ibm.ws.springboot.fat20.webanno.app
-- rename of com.ibm.ws.springboot.support.version20.test.webanno.app
com.ibm.ws.springboot.fat20.webflux.app
-- rename of com.ibm.ws.springboot.support.version20.test.webflux.app
com.ibm.ws.springboot.fat20.websocket.app
-- rename of com.ibm.ws.springboot.support.version20.test.websocket.app
com.ibm.ws.springboot.fat20_fat
-- rename of com.ibm.ws.springboot.support_fat

com.ibm.ws.springboot.fat30.actuator.app
-- copied from com.ibm.ws.springboot.fat20.actuator.app
com.ibm.ws.springboot.fat30.app
-- copied from com.ibm.ws.springboot.fat20.app
com.ibm.ws.springboot.fat30.java.app
-- copied from com.ibm.ws.springboot.fat20.java.app
com.ibm.ws.springboot.fat30.multicontext.app
-- copied from com.ibm.ws.springboot.fat20.multicontext.app
com.ibm.ws.springboot.fat30.war.app
-- copied from com.ibm.ws.springboot.fat20.war.app
com.ibm.ws.springboot.fat30.webanno.app
-- copied from com.ibm.ws.springboot.fat20.webanno.app
com.ibm.ws.springboot.fat30.webflux.app
-- copied from com.ibm.ws.springboot.fat20.webflux.app
com.ibm.ws.springboot.fat30.websocket.app
-- copied from com.ibm.ws.springboot.fat20.websocket.app
com.ibm.ws.springboot.fat30_fat
-- copied from com.ibm.ws.springboot.fat20_fat

dev/com.ibm.ws.app.manager.springboot/.classpath
dev/com.ibm.ws.springboot.support.shutdown/.classpath
dev/com.ibm.ws.springboot.support.web.server/.classpath
dev/com.ibm.ws.springboot.support.web.server.version15/.classpath
dev/com.ibm.ws.springboot.support.web.server.version20/.classpath
dev/com.ibm.ws.springboot.utility/.classpath
dev/com.ibm.ws.springboot.utility.springbootutil/.classpath
dev/io.openliberty.springboot.support.web.server.version30/.classpath
-- Unify eclipse output with gradle build output.

X dev/com.ibm.ws.app.manager.springboot/.gitignore
X dev/com.ibm.ws.springboot.support.shutdown/.gitignore
X dev/com.ibm.ws.springboot.support.web.server.version15/.gitignore
X dev/com.ibm.ws.springboot.utility/.gitignore
X dev/com.ibm.ws.springboot.utility.springbootutil/.gitignore
-- No longer needed w/unified outputs.

X dev/com.ibm.ws.springboot.support.shutdown/test/readme
X dev/com.ibm.ws.springboot.support.web.server.version15/test/readme
X dev/com.ibm.ws.springboot.support.web.server.version20/test/readme
X dev/com.ibm.ws.springboot.support.web.server/test/readme
-- Obsolete; this project has no unit tests.

tbitonti committed:
Initial Spring 30 FAT update;
Creation of 30 FAT as copies of 20 FATs.
Several test driven fixes;
FeatureAuditor updates for SB3 cases (still in progress).
Partially unified spring version dependencies;
.classpath and .gitignore cleanup.
Added SB3 server config template.

hlhoots reviewed (dismissed):
LGTM pending a good build

tbitonti committed:
Re-add springBoot3 template.  Comment out failing tests.  Added micro.version note to LibertyServer.

tbitonti commented:
#build

tbitonti commented:
All tests of interest passed:
```
79 0 0 com.ibm.ws.springboot.fat20_fat FAT tests
68 0 0 com.ibm.ws.springboot.fat30_fat FAT tests
47 0 0 com.ibm.ws.springboot.utility unittests
```

cbridgha reviewed (dismissed):
Looks good

tbitonti committed:
Spring FAT adjustments.

tbitonti commented:
#build

tbitonti committed:
Re-enable Spring 20 FAT tests.

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti committed:
FeatureAuditor updates.
Spring FAT 20 and FAT 30 fixes.

tbitonti commented:
#build

jhanders34 commented on dev/build.image/publish/wlp/templates/servers/springBoot3/server.xml:
```diff
@@ -0,0 +1,24 @@
+
+    <!-- Enable features -->
+    <featureManager>
+        <feature>servlet-6.0</feature>
+        <feature>pages-3.1</feature>
+        <feature>webSocket-2.1</feature>
+
+        <feature>transportSecurity-1.0</feature>
+
+        <feature>springBoot-3.0</feature>
```
```suggestion
        <feature>springBoot-3.0</feature>
        <feature>servlet-6.0</feature>
        <feature>pages-3.1</feature>
        <feature>webSocket-2.1</feature>
        <feature>transportSecurity-1.0</feature>
```

This is consistent with the springBoot1 and springBoot2 templates.

tbitonti replied:
Re: The missing servlet and websockets tests not working doesn't make sense to me. It would seem that the started server has features in it still for some reason.
I was able to get those working.  I had problems in my update to FeatureAuditor.

tbitonti replied:
Re: Messaging changes.  I have the same worry.  The difficulty here is that the existing messages don't work very well when adding in the spring3 tests.  Also, the existing messages are not quite accurate.

tbitonti replied:
ok, the sprint2 broken missing servlet and missing websocket tests are working.  What remains broken are the spring3 missing servlet and missing websocket tests.  What am seeing is that jakarta Servlet and jakarta WebSocket are loadable with just the springBoot-3.0 feature set.  That means the condition for emitting the missing feature message is never triggerred, and the tests fail.

jhanders34 commented on dev/cnf/resources/bnd/bundle.spring.props:
```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
```suggestion
```

Not needed in new files.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/.classpath:
```diff
@@ -1,8 +1,8 @@
-	<classpathentry kind="src" output="bin" path="src"/>
-	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
+	<classpathentry kind="src" output="build/classes/java/test" path="test"/>
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
These changes should be reverted.  For eclipse output we use bin and bin_test.  For gradle we use build/classes...

tbitonti replied:
Easy enough to change these back.  But, having the paths be the same for eclipse and gradle not work?  That removes .ignore entries, and simplifies cleaning up.

jhanders34 replied:
The gradle and eclipse output directories are purposely in different directories for every projects in open liberty.  This allows for both to be able to build at the same time without interfering with each other.  The .gitignore files in the project directories can mostly be removed because the root level .gitignore file already has `/*/*/bin/` and `/*/*/build/`

tbitonti replied:
Fixed.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -99,8 +104,63 @@ error.application.not.found.inside.packaged.liberty.jar=CWWKC0266E: The applicat
+#0: The current Java version
+#1: The provisioned Spring Boot version.
+#2: The Java version required for the provisioned Java version
+warning.unsupported.spring.java.version=CWWKC0270W: The current Java version {0} does not support the provisioned Spring Boot version {1}: Java version {2} or higher is required.
```
I believe ID usually states to use `later` instead of `higher`, but ID should be reviewing this message files.

tbitonti replied:
Updated to remove the "or higher" text.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -99,8 +104,63 @@ error.application.not.found.inside.packaged.liberty.jar=CWWKC0266E: The applicat
+warning.unsupported.spring.java.version.explanation=Spring Boot has specific Java version requirements.  The current Java version does not support the provisioned Spring Boot version.
+warning.unsupported.spring.java.version.useraction=Either, change the Java version to a required version, or change to a different Spring Boot version.
+
+error.spring3.required=CWWKC0273E: Error: Feature springBoot-3.0 (or higher) must be provisioned: Feature springBoot-1.5 or springBoot-2.0 is provisioned and the application has Spring 3.0 content.
```
Not sure we should say `(or higher)` here.  It may not be accurate for a later version of springBoot to work with SpringBoot 3.0 content.

tbitonti replied:
Yeah.  Generally, these messages a problematic, in that if and when there is a spring 4 they will need to be updated again.  There are problems, too, in that there are finer version dependencies than we are checking.

tbitonti replied:
The messages have been updated to remove the (or higher) text, and to reassign the numbers back to the spring2 values.  The messages have the same meanings as before.  In one case, there are two messages which became the same once they were parameterized.  I left the two messages and two original message codes, even though one of them is now redundant.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -99,8 +104,63 @@ error.application.not.found.inside.packaged.liberty.jar=CWWKC0266E: The applicat
+error.spring3.requires.servlet6.application.explanation=The application uses Servlets but a Servlet feature is not provisioned.
+error.spring3.requires.servlet6.application.useraction=Add the indicated Servlet feature.
+
+error.spring3.requires.websocket2.application=CWWKC0275E: Error: Feature websocket-2.0 (or higher) must be provisioned: Feature springBoot-3.0 is provisioned and the application uses WebSockets.
```
websocket-2.1 is the EE 10 feature so the version should be 2.1 not 2.0.  Again I don't think `(or higher)` should be included in this message.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -99,8 +104,63 @@ error.application.not.found.inside.packaged.liberty.jar=CWWKC0266E: The applicat
+error.spring3.requires.websocket2.application.explanation=The application uses Web Sockets but a Web Socket feature is not provisioned.
+error.spring3.requires.websocket2.application.useraction=Add the indicated Web Socket feature.
+
+error.spring2.required=CWWKC0278E: Error: Feature springBoot-1.5 or springBoot-2.0 must be provisioned: Feature springBoot-3.0 (or higher) is provisioned and the application has Spring 1.5 or Spring 2.0 content.
```
Again I think `(or higher)` should be removed from this message.

jhanders34 commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -99,8 +104,63 @@ error.application.not.found.inside.packaged.liberty.jar=CWWKC0266E: The applicat
+error.spring2.requires.websocket1.application.explanation=The application uses Web Sockets but a Web Socket feature is not provisioned.
+error.spring2.requires.websocket1.application.useraction=Add one of the indicated Web Socket feature.
+
+# Not in use; no translation required.
```
Why add them here if they are not used?

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.actuator.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
<classpath>
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/cnf/oss_dependencies.maven:
```diff
@@ -766,42 +766,58 @@ org.slf4j:slf4j-jdk14:1.7.36
 org.springframework.boot:spring-boot-autoconfigure:1.5.9.RELEASE
-org.springframework.boot:spring-boot-autoconfigure:2.0.0.RELEASE
-org.springframework.boot:spring-boot-autoconfigure:2.7.1
+org.springframework.boot:spring-boot-autoconfigure:2.2.5.RELEASE
```
Why 2.2.5 instead of 2.7.1?  I see in the FAT projects you are using 2.7.1 in the gradle files, but 2.2.5 in the bnd.bnd file.  Why the mismatch?

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.app/.project:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-    <name>com.ibm.ws.springboot.support.version20.test.app</name>
+    <name>com.ibm.ws.springboot.fat20.app</name>
```
Why the rename?  It seems like you are renaming just to rename it.  If you are going to rename, you might as well change from com.ibm.ws to io.openliberty prefix.  Same for other project renames.
```suggestion
    <name>io.openliberty.springboot.fat20.app</name>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.actuator.app/bnd.bnd:
```diff
@@ -13,22 +13,12 @@
-	com.ibm.websphere.javaee.servlet.3.1
+src: \
+  src/main/java, \
+  src/main/resources
```
Why doesn't this have a -testpath and look like the fat20.app bnd.bnd file?

tbitonti replied:
Dropped during the updates.  I'm putting -testpath back in.

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.app/build.gradle:
```diff
@@ -0,0 +1,57 @@
+  implementation('org.springframework.security:spring-security-core' + ':' + sv['springSecurity.fat'])
+  implementation('org.springframework.security:spring-security-web' + ':' + sv['springSecurity.fat'])
+
+  // Why multiple versions of the same dependency??
```
That was my same question.

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.java.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.multicontext.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.war.app/.classpath:
```diff
@@ -0,0 +1,10 @@
+  <classpathentry kind="src" path="module/src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="src" path="module/src/main/java" output="bin"/>
  <classpathentry kind="src" path="module/src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.webanno.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.webflux.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20.websocket.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20_fat/.classpath:
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" path="fat/src"/>
+	<classpathentry kind="src" path="fat/src" output="build/classes/java/main"/>
```
```suggestion
	<classpathentry kind="src" path="fat/src" output="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20_fat/.classpath:
```diff
@@ -1,7 +1,7 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build/classes"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat20_fat/.project:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <projectDescription>
-	<name>com.ibm.ws.springboot.support_fat</name>
+	<name>com.ibm.ws.springboot.fat20_fat</name>
```
This tests both 1.5 and 2.0.  Why is it called fat20_fat?

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/.project:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.ibm.ws.springboot.fat30.actuator.app</name>
```
Since it is a new project it should start with io.openliberty
```suggestion
	<name>io.openliberty.springboot.fat30.actuator.app</name>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+eclipse.preferences.version=1
+org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
+org.eclipse.jdt.core.compiler.codegen.methodParameters=do not generate
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.8
+org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
+org.eclipse.jdt.core.compiler.compliance=1.8
+org.eclipse.jdt.core.compiler.debug.lineNumber=generate
+org.eclipse.jdt.core.compiler.debug.localVariable=generate
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

tbitonti replied:
Updated

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17/"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/.project:
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>com.ibm.ws.springboot.fat30.app</name>
```
```suggestion
	<name>io.openliberty.springboot.fat30.app</name>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/build.gradle:
```diff
@@ -0,0 +1,65 @@
+  // 0.46
+  // https://mvnrepository.com/artifact/org.webjars/jquery
+  // 3.6.4
+
```
```suggestion
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/src/main/java/com/ibm/ws/springboot/fat30/actuator/app/TestApplication.java:
```diff
@@ -0,0 +1,32 @@
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.springboot.fat30.actuator.app;
```
```suggestion
package io.openliberty.springboot.fat30.actuator.app;
```
Since it is a new project, it should have io.openliberty package.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/src/main/resources/templates/login.html:
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:th="http://www.thymeleaf.org"
+      xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity3">
```
Should these be `https` instead of `http`?

tbitonti replied:
I can't say.  I added a comment to this (and to "hello.html") in fat20.app and fat30.app to preserve this question.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.java.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.java.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.multicontext.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+    <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+    <classpathentry kind="output" path="build"/>
```
```suggestion
    <classpathentry kind="src" path="src/main/java" output="bin"/>
    <classpathentry kind="src" path="src/main/resources" output="bin"/>
    <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
    <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
    <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.multicontext.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.java.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/.project:
```diff
@@ -0,0 +1,29 @@
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

tbitonti replied:
Updated

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.multicontext.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/.classpath:
```diff
@@ -0,0 +1,10 @@
+  <classpathentry kind="src" path="module/src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="src" path="module/src/main/java" output="bin"/>
  <classpathentry kind="src" path="module/src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/module/build.gradle:
```diff
@@ -0,0 +1,9 @@
+
+group = 'com.ibm.ws.example'
+version = '0.0.1-SNAPSHOT'
+sourceCompatibility = 1.8
```
```suggestion
sourceCompatibility = 17
```

tbitonti replied:
Fixed.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/module/src/main/resources/application-context-module.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
```
should these be `https` instead of `http`?

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/src/main/webapp/WEB-INF/jsp/testjsp.jsp:
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags"%>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core"%>
```
Shouldn't this be jakarta?

tbitonti replied:
Hmm.  It seems that it should be.  However, these show the same URI:
https://jakarta.ee/specifications/tags/1.2/tagdocs/c/tld-summary.html
https://jakarta.ee/specifications/tags/2.0/tagdocs/c/tld-summary.html

jhanders34 replied:
EE 10 though is Jakarta Tags 3.0, so https://jakarta.ee/specifications/tags/3.0/tagdocs/c/tld-summary.html is the right reference.  In that reference the URI is `jakarta.tags.core`

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/bnd.bnd:
```diff
@@ -0,0 +1,48 @@
+  org.springframework:spring-web;${springVersion30}, \
+\
+  io.openliberty.jakarta.servlet.6.0
+
+// io.openliberty.jakartaee-10
+// com.ibm.websphere.appserver.servlet-6.0
```
```suggestion
```

tbitonti replied:
Removed.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/build.gradle:
```diff
@@ -0,0 +1,31 @@
+
+group = 'com.ibm.ws.example.spring'
+version = '0.0.1-SNAPSHOT'
+sourceCompatibility = 1.8
```
```suggestion
sourceCompatibility = 17
```

tbitonti replied:
Fixed.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webflux.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webflux.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webflux.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.actuator.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

tbitonti replied:
Removed.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.java.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.multicontext.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.war.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webanno.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.webflux.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.websocket.app/.classpath:
```diff
@@ -0,0 +1,8 @@
+  <classpathentry kind="src" path="src/main/resources" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8/"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src/main/java" output="bin"/>
  <classpathentry kind="src" path="src/main/resources" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.websocket.app/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.websocket.app/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30.websocket.app/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/.classpath:
```diff
@@ -0,0 +1,7 @@
+	<classpathentry kind="src" path="fat/src" output="build/classes/java/main"/>
+	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
+	<classpathentry kind="output" path="build/classes"/>
```
```suggestion
	<classpathentry kind="src" path="fat/src" output="bin"/>
	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/.project:
```diff
@@ -0,0 +1,29 @@
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>bndtools.core.bndnature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
```
```suggestion
	</buildSpec>
	<natures>
		<nature>org.eclipse.jdt.core.javanature</nature>
		<nature>bndtools.core.bndnature</nature>
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/.settings/org.eclipse.buildship.core.prefs:
```diff
@@ -0,0 +1,2 @@
+connection.project.dir=
```
delete this file

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,290 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=17
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=17
```

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/bnd.bnd:
```diff
@@ -0,0 +1,96 @@
+src: \
+  fat/src
+
+// websocket 2.1 is a small overlay on websocket 2.0;
+// putting websocket 2.1 on the build path is not sufficient;
+// websocket 2.0 must also be there
+
+-buildpath: \
+  io.openliberty.org.apache.commons.logging;version=latest,\
+  io.openliberty.org.apache.commons.codec;version=latest,\
+  io.openliberty.org.apache.xercesImpl;version=latest, \
+\
+  net.sourceforge.htmlunit:htmlunit;version=2.44.0,\
+  net.sourceforge.htmlunit:htmlunit-cssparser;version=1.6.0,\
+  net.sourceforge.htmlunit:neko-htmlunit;strategy=exact;version=2.44.0,\
+  net.sourceforge.htmlunit:webdriver;version=2.6,\
+  org.seleniumhq.webdriver:webdriver-common;version=0.9.7376,\
+  org.brotli:dec;version=0.1.2,\
+\
+  io.openliberty.jakarta.servlet.6.0;version=latest, \
+  io.openliberty.jakarta.websocket.2.0;version=latest, \
+  io.openliberty.jakarta.websocket.2.1;version=latest, \
```
```suggestion

-buildpath: \
  io.openliberty.org.apache.commons.logging;version=latest,\
  io.openliberty.org.apache.commons.codec;version=latest,\
  io.openliberty.org.apache.xercesImpl;version=latest, \
\
  net.sourceforge.htmlunit:htmlunit;version=2.44.0,\
  net.sourceforge.htmlunit:htmlunit-cssparser;version=1.6.0,\
  net.sourceforge.htmlunit:neko-htmlunit;strategy=exact;version=2.44.0,\
  net.sourceforge.htmlunit:webdriver;version=2.6,\
  org.seleniumhq.webdriver:webdriver-common;version=0.9.7376,\
  org.brotli:dec;version=0.1.2,\
\
  io.openliberty.jakarta.servlet.6.0;version=latest, \
  io.openliberty.jakarta.websocket.2.1;version=latest, \
  io.openliberty.jakarta.websocket.client.2.1;version=latest, \
```

tbitonti replied:
Updated.

jhanders34 commented on dev/com.ibm.ws.springboot.fat30_fat/fat/src/com/ibm/ws/springboot/support/fat/CommonWebFluxTests30.java:
```diff
@@ -0,0 +1,60 @@
+    public Map<String, String> getBootStrapProperties() {
+        Map<String, String> properties = new HashMap<>();
+        // add channel trace for RTC defect 266559
+        properties.put("com.ibm.ws.logging.trace.specification",
```
Should we not start with trace enabled since this is a new test?

jhanders34 commented on dev/com.ibm.ws.springboot.support.shutdown/.classpath:
```diff
@@ -1,8 +1,7 @@
-	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+  <classpathentry kind="src" path="src" output="build/classes/java/main"/>
+  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
+  <classpathentry kind="output" path="build"/>
```
```suggestion
  <classpathentry kind="src" path="src" output="bin"/>
  <classpathentry kind="con" path="aQute.bnd.classpath.container"/>
  <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
  <classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server.version15/.classpath:
```diff
@@ -1,8 +1,7 @@
 <classpath>
-	<classpathentry kind="src" output="bin" path="src"/>
-	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
```
```suggestion
	<classpathentry kind="src" output="bin" path="src"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server.version15/.classpath:
```diff
@@ -1,8 +1,7 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server.version20/.classpath:
```diff
@@ -1,8 +1,7 @@
 <classpath>
-	<classpathentry kind="src" output="bin" path="src"/>
-	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
```
```suggestion
	<classpathentry kind="src" output="bin" path="src"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server.version20/.classpath:
```diff
@@ -1,8 +1,7 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server/.classpath:
```diff
@@ -1,8 +1,7 @@
 <classpath>
-	<classpathentry kind="src" output="bin" path="src"/>
-	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
```
```suggestion
	<classpathentry kind="src" output="bin" path="src"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.support.web.server/.classpath:
```diff
@@ -1,8 +1,7 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.utility.springbootutil/.classpath:
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
+	<classpathentry kind="src" path="src" output="build/classes/java/main"/>
```
```suggestion
	<classpathentry kind="src" path="src" output="bin"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.utility.springbootutil/.classpath:
```diff
@@ -1,6 +1,7 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

tbitonti replied:
These have all been changed back.

jhanders34 commented on dev/com.ibm.ws.springboot.utility/.classpath:
```diff
@@ -1,8 +1,8 @@
 <classpath>
-	<classpathentry kind="src" output="bin" path="src"/>
-	<classpathentry kind="src" output="bin_test" path="test"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
+	<classpathentry kind="src" output="build/classes/java/test" path="test"/>
```
```suggestion
	<classpathentry kind="src" output="bin" path="src"/>
	<classpathentry kind="src" output="bin_test" path="test"/>
```

jhanders34 commented on dev/com.ibm.ws.springboot.utility/.classpath:
```diff
@@ -1,8 +1,8 @@
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 commented on dev/io.openliberty.springboot.support.web.server.version30/.classpath:
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
-	<classpathentry kind="src" output="bin" path="src"/>
+	<classpathentry kind="src" output="build/classes/java/main" path="src"/>
```
```suggestion
	<classpathentry kind="src" output="bin" path="src"/>
```

jhanders34 commented on dev/io.openliberty.springboot.support.web.server.version30/.classpath:
```diff
@@ -1,7 +1,7 @@
-	<classpathentry kind="output" path="bin"/>
-</classpath>
\ No newline at end of file
+	<classpathentry kind="output" path="build"/>
```
```suggestion
	<classpathentry kind="output" path="bin"/>
```

jhanders34 reviewed (dismissed):
In general the com.ibm.ws to io.openliberty comment applies to the projects that you modified the name or are new projects.  I only put the comment once.

For SpringBoot 3 testing I think you need to move to a new version of jetty.  See io.openliberty.wsoc.2.1.internal_fat/build.gradle for updated version numbers of jetty websockets for instance.

The missing servlet and websockets tests not working doesn't make sense to me.  It would seem that the started server has features in it still for some reason.

I am concerned about the messaging changes.  Is putting out a new message for the same error condition considered a zero migration breaking change?

tbitonti committed:
More test fixes.

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti committed:
First pass corrections per Jared's review.

tbitonti committed:
Pass 2 over test updates per review.

tbitonti committed:
FAT adjustments.  Locally, all tests run except for five classes:
missing servlets; missing websockets;
websockets; trace; security
These test classes are currently disabled.

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti committed:
Misc fixes per review:
Fixed missing change from Java 1.8 to Java 17.
Fixed .project and .settings files.
Renamed spring30 projects to io.openliberty.
Added missing -testpath to actuator FAT projects.
Removed extraneous comments from several files.
Corrected jstl taglib URI.
Commented on "http" vs "https" URIs in hello.html and login.html, but did not change the URIs.
Left "spring20" as the name for 1.5 and 2.0 FATS, for historical continuity.

tbitonti commented:
#build

tbitonti commented:
#build

tbitonti committed:
Update dev/build.image/publish/wlp/templates/servers/springBoot3/server.xml

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

tbitonti committed:
Update dev/cnf/resources/bnd/bundle.spring.props

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

tbitonti commented:
#build

tbitonti committed:
Add back application error.

tbitonti commented:
#build

tbitonti commented:
100% green for the enabled spring FATs.

hlhoots reviewed (dismissed):
LGTM pending and ID review of message additions/updates

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#0: The required spring feature.
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
```
what is "provisioned" in this context? We generally talk about features as being "enabled" or "configured". Unless "provisioned" has specific meaning, I'd recommend using "configured". Otherwise, need to explain what "provisioned" means in this context

tbitonti replied:
Changing "provisioned" to "configured".  (I've been using "provisioned" to mean listed as a feature in the server configuration.)

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
+error.spring.required.15.explanation=The Spring Boot feature is not compatible with application content.
```
```suggestion
error.spring.required.15.explanation=The currently configured Spring Boot feature is not compatible with application content.
```

tbitonti replied:
Added.

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
+error.spring.required.15.explanation=The Spring Boot feature is not compatible with application content.
+error.spring.required.15.useraction=Change the Spring Boot feature to be compatible with application content.
```
```suggestion
error.spring.required.15.useraction=Configure a version of the Spring Boot feature that is compatible with application content.
```

tbitonti replied:
Updated.

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#0: The required spring feature.
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.20=CWWKC0253E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
```
see previous comments for lines 48-50 for lines 60-62

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#0: The required spring feature.
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
```
I don't think  "Error:" needs to be appended to the message because the message code ends in "E"

tbitonti replied:
Removed "Error:" from this and other messages.

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#0: The required spring feature.
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: Error: Feature {0} must be provisioned: Feature {1} is provisioned and the application has Spring {2} content.
```
```suggestion
error.spring.required.15=CWWKC0252E: The {0} feature must be configured in the server.xml file. The {1} feature  is configured and the application has Spring {2} content.
```

tbitonti replied:
Updated.

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+error.spring.required.20.useraction=Change the Spring Boot feature to be compatible with application content.
+
+#ADDED for SpringBoot 3.0:
+error.spring3.required=CWWKC0273E: Error: Feature springBoot-3.0 must be provisioned: Feature springBoot-1.5 or springBoot-2.0 is provisioned and the application has Spring 3.0 content.
```
see previous comments

dmuelle reviewed (changes requested):
ID review- I think the initial suggestions I made, if correct can be applied to each of the subsequent message sets. Let me know if you have any questions or if I've misunderstood anything. Thanks!

tbitonti committed:
Message updates per review.

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#0: The required spring feature.
+#1: The current spring feature.
+#2: The version of spring content found in the application.
+error.spring.required.15=CWWKC0252E: The {0} feature must be configured in the server.xml file. The {1} feature  is configured and the application has Spring {2} content.
```
```suggestion
error.spring.required.15=CWWKC0252E: The {0} feature must be configured in the server.xml file. The {1} feature is configured and the application has Spring {2} content.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+error.spring.required.20.useraction=Configure a version of the Spring Boot feature that is compatible with application content.
+
+#ADDED for SpringBoot 3.0:
+error.spring3.required=CWWKC0273E: Feature springBoot-3.0 must be configured: Feature springBoot-1.5 or springBoot-2.0 is configured and the application has Spring 3.0 content.
```
```suggestion
error.spring3.required=CWWKC0273E: The springBoot-3.0 feature must be configured. The springBoot-1.5 or springBoot-2.0 feature is configured and the application has Spring 3.0 content.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#error.missing.servlet.explanation=The application is a web application, and it requires a servlet feature to be configured in the server.xml file.
+#error.missing.servlet.useraction=Configure a servlet feature in the server.xml file.
+
+error.spring2.requires.servlet31.application=CWWKC0254E: Feature servlet-3.1 or servlet-4.0 must be configured: Feature springBoot-1.5 or springBoot-2.0 is configured and the application uses Servlets.
```
```suggestion
error.spring2.requires.servlet31.application=CWWKC0254E: The servlet-3.1 or servlet-4.0 feature  must be configured. The springBoot-1.5 or springBoot-2.0 feature is configured and the application uses servlets.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#error.missing.servlet.useraction=Configure a servlet feature in the server.xml file.
+
+error.spring2.requires.servlet31.application=CWWKC0254E: Feature servlet-3.1 or servlet-4.0 must be configured: Feature springBoot-1.5 or springBoot-2.0 is configured and the application uses Servlets.
+error.spring2.requires.servlet31.application.explanation=The application uses Servlets but a Servlet feature is not configured.
```
- ```suggestion

error.spring2.requires.servlet31.application.explanation=The application uses servlets but no Servlet feature is configured.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+
+error.spring2.requires.servlet31.application=CWWKC0254E: Feature servlet-3.1 or servlet-4.0 must be configured: Feature springBoot-1.5 or springBoot-2.0 is configured and the application uses Servlets.
+error.spring2.requires.servlet31.application.explanation=The application uses Servlets but a Servlet feature is not configured.
+error.spring2.requires.servlet31.application.useraction=Configure one of the indicated Servlet features.
```
```suggestion
error.spring2.requires.servlet31.application.useraction=Configure one of the specified Servlet features.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+error.spring2.requires.servlet31.application.useraction=Configure one of the indicated Servlet features.
+
+#ADDED for SpringBoot 3.0:
+error.spring3.requires.servlet6.application=CWWKC0274E: Feature servlet-6.0 must be configured: Feature springBoot-3.0 is configured and the application uses Servlets.
```
```suggestion
error.spring3.requires.servlet6.application=CWWKC0274E: The servlet-6.0 feature must be configured. The springBoot-3.0 feature is configured and the application uses servlets.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+
+#ADDED for SpringBoot 3.0:
+error.spring3.requires.servlet6.application=CWWKC0274E: Feature servlet-6.0 must be configured: Feature springBoot-3.0 is configured and the application uses Servlets.
+error.spring3.requires.servlet6.application.explanation=The application uses Servlets but a Servlet feature is not configured.
```
```suggestion
error.spring3.requires.servlet6.application.explanation=The application uses servlets but no Servlet feature is configured.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+#ADDED for SpringBoot 3.0:
+error.spring3.requires.servlet6.application=CWWKC0274E: Feature servlet-6.0 must be configured: Feature springBoot-3.0 is configured and the application uses Servlets.
+error.spring3.requires.servlet6.application.explanation=The application uses Servlets but a Servlet feature is not configured.
+error.spring3.requires.servlet6.application.useraction=Configure the indicated Servlet feature.
```
```suggestion
error.spring3.requires.servlet6.application.useraction=Configure the specified Servlet feature.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+
+#UPDATED for SpringBoot 3.0:
+#error.missing.servlet=CWWKC0254E: The application failed to start because it requires a servlet feature to be configured in the server.xml file.
+#error.missing.servlet.explanation=The application is a web application, and it requires a servlet feature to be configured in the server.xml file.
```
```suggestion
#error.missing.servlet.explanation=The application is a web application, and it requires a Servlet feature to be configured in the server.xml file.
```

dmuelle commented on dev/com.ibm.ws.app.manager.springboot/resources/com/ibm/ws/app/manager/springboot/internal/resources/Messages.nlsprops:
```diff
@@ -37,17 +37,48 @@ warning.could.not.thin.application=CWWKC0251W: Could not convert the application
+error.spring3.required.useraction=Configure a version of the Spring Boot feature that is compatible with application content.
+
+#UPDATED for SpringBoot 3.0:
+#error.missing.servlet=CWWKC0254E: The application failed to start because it requires a servlet feature to be configured in the server.xml file.
```
```suggestion
#error.missing.servlet=CWWKC0254E: The application failed to start because it requires a Servlet feature to be configured in the server.xml file.
```

dmuelle reviewed (changes requested):
ID review- a few more minor edits

tbitonti committed:
Additional message updates.

dmuelle reviewed (approved):
LGTM. thanks

tbitonti commented:
#build

tbitonti commented:
#libby

tbitonti commented:
The last build had no failures associated with this PR.
